CRM/Contribute/Form/Contribution.php

```diff
@@ -1021,6 +1046,23 @@ public static function formRule($fields, $files, $self) {
       $errors['financial_type_id'] = ' ';
       $errors['_qf_default'] = $e->getMessage();
     }
+    if (!$self->_mode && !empty($fields['pan_truncation'])
+    ) {
+      $contributionDoneViaPaymentProcesor = FALSE;
+      if ($self->_id) {
+        $financialTrxn = $self->getlatestPayments();
+        if (!empty($financialTrxn['financial_trxn_id.payment_processor_id'])) {
+          $contributionDoneViaPaymentProcesor = TRUE;
+        }
+      }
+      if (!$contributionDoneViaPaymentProcesor
+        && (!is_numeric($fields['pan_truncation'])
+          || strlen($fields['pan_truncation']) != 4
+        )
+      ) {
+        $errors['pan_truncation'] = ts('Please enter valid last 4 digit card number.');
```

Comment:
No, I think it says, 'if this was not done online and (a non-numeric PAN is entered or not 4 digits of the PAN is entered) then display error.